tests/environment.py

```diff
@@ -0,0 +1,175 @@
+###############################################################
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+# Copyright (c) 2020 Arnout Vandecappelle (Essensium/Mind)
+# This code is subject to the terms of the BSD+Patent license.
+# See LICENSE file for more details.
+###############################################################
+
+import json
+import os
+import re
+import subprocess
+
+from capi import UCCSocket
+from opts import debug
+import sniffer
+
+
+class AlEntity:
+    '''Abstract representation of a MultiAP device (1905.1 AL Entity).
+
+    Derived classes provide concrete implementations for a specific device (e.g. docker).
+
+    This provides basic information about the entity, e.g. its AL MAC address. How this information
+    is retrieved is implementation-specific.
+
+    It also provides an abstract interface to interact with the entity, e.g. for sending CAPI
+    commands.
+
+    If a device runs both the agent and the controller, two AlEntities should be created for it,
+    with the same MAC address. This is not how it is modeled in 1905.1, but it corresponds to how
+    it is implemented in prplMesh and it allows us to have e.g. a separate UCCSocket to controller
+    and agent.
+    '''
+    def __init__(self, mac: str, ucc_socket: UCCSocket, is_controller: bool = False):
+        self.mac = mac
+        self.ucc_socket = ucc_socket
+        self.is_controller = is_controller
+        self.radios = []
+
+        # Convenience functions that propagate to ucc_socket
+        self.cmd_reply = self.ucc_socket.cmd_reply
+        self.dev_get_parameter = self.ucc_socket.dev_get_parameter
+        self.dev_send_1905 = self.ucc_socket.dev_send_1905
+
+    def command(self, *command: str) -> bytes:
```

Comment:
Please specify the type of command that should be sent with this method.
Also, examples in the docstring would be helpful
e.g.
```
    def sendCAPIcommand(self, *command: str) -> bytes:
    '''Send the specified CAPI command to the device.
    The command is a variable length argument list comprising the different words of the command, e.g.
    device.sendCAPIcommand("reset_device")
    device.sendCAPIcommand("configure_device", "00:11:22:33:44:55")
```

Reply:
Actually, I notice now that this interface is for CLI commands. Still, the same principle applies :)